src/server/game/Entities/Player/Player.cpp

```diff
@@ -1081,6 +1081,11 @@ void Player::Update(uint32 p_time)
 
     UpdateAfkReport(now);
 
+    if (IsInCombat())
```

Comment:
Should only check for pvp combat. you can attack an enemy player, then engage an npc and if you wait long enough your pvp talents will be disabled while you're still in combat with npc. Is there a way to check this ?
Also I have seen this buff on npc's so I will have to move it to Unit and check if the behaviour is the same